lvk/LVK.cpp

```diff
@@ -181,20 +181,32 @@ void lvk::destroy(lvk::IContext* ctx, lvk::TextureHandle handle) {
 // Logs GLSL shaders with line numbers annotation
 void lvk::logShaderSource(const char* text) {
   uint32_t line = 1;
-
-  LLOGL("\n(%3u) ", line);
+  
+  std::string source;
+  source.reserve(128);
+  constexpr uint32_t kBufferLength = 16;
+  char buffer[kBufferLength];
+  
+  snprintf(buffer, kBufferLength, "\n(%3u) ", line);
+  source.append(buffer);
 
   while (text && *text) {
     if (*text == '\n') {
-      LLOGL("\n(%3u) ", ++line);
+      source.append("\n");
+      LLOGL(source.c_str());
+      source.clear();
+      
+      snprintf(buffer, kBufferLength, "(%3u) ", ++line);
+      source.append(buffer);
     } else if (*text == '\r') {
       // skip it to support Windows/UNIX EOLs
     } else {
-      LLOGL("%c", *text);
+      source.push_back(*text);
     }
     text++;
   }
-  LLOGL("\n");
+  source.append("\n");
+  LLOGL(source.c_str());
```

Comment:
Something like this should work with HTML (but it does not work well with plain text output). I suggest making two different function and using them based on if Minilog has HTML output enabled.

```
void lvk::logShaderSource(const char* text) {
  uint32_t line = 1;

  LLOGL("\n(%3u) ", line);

  uint32_t numChars = 0;
  const char* lineStart = text;

  while (text && *text) {
    if (*text == '\n') {
      if (numChars) {
        LLOGL("\n(%3u)%.*s", ++line, numChars, lineStart);
      } else {
        LLOGL("\n(%3u)", ++line);
		}
      numChars = 0;
      lineStart = text + 1;
    } else if (*text == '\r') {
      // skip it to support Windows/UNIX EOLs
      numChars = 0;
      lineStart = text + 1;
    } else {
      numChars++;
    }
    text++;
  }
  if (numChars) {
    LLOGL("\n(%3u)%.*s", ++line, numChars, lineStart);
  }
  LLOGL("\n");
}
```